From b984c50b630f3f3c29fd918a5c1858b65df5e382 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 13 Mar 2024 15:52:30 +0100
Subject: [PATCH 0721/1695] drivers: stm32_gpio: rename stm32_gpio_pm()

Rename stm32_gpio_pm() functions and the helper functions it calls
to prevent confusion with the newly introduced PM callback
consumed_gpios_pm(). Since stm32_gpio_pm() only deals with the
pin muxing security state configuration, rename stm32_gpio_pm()
to stm32_gpio_sec_config_pm() and use equivalent changes for
its local helper functions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 4ee73453aef1ec2193cc31cbb2a6f3a5907025a4 drivers: stm32_gpio: rename stm32_gpio_pm()]
Change-Id: Iba43213b752da90248f3cc773b6381c835dd1c85
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/366524
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_gpio.c | 25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

diff --git a/core/drivers/stm32_gpio.c b/core/drivers/stm32_gpio.c
index eedd34c46..50adcb3ac 100644
--- a/core/drivers/stm32_gpio.c
+++ b/core/drivers/stm32_gpio.c
@@ -1241,7 +1241,7 @@ static void stm32_gpio_set_conf_sec(struct stm32_gpio_bank *bank)
 	}
 }
 
-static TEE_Result stm32_gpio_pm_resume(void)
+static TEE_Result stm32_gpio_sec_config_resume(void)
 {
 	struct stm32_gpio_bank *bank = NULL;
 
@@ -1267,7 +1267,7 @@ static TEE_Result stm32_gpio_pm_resume(void)
 	return TEE_SUCCESS;
 }
 
-static TEE_Result stm32_gpio_pm_suspend(void)
+static TEE_Result stm32_gpio_sec_config_suspend(void)
 {
 	struct stm32_gpio_bank *bank = NULL;
 
@@ -1289,23 +1289,23 @@ static TEE_Result stm32_gpio_pm_suspend(void)
 }
 
 static TEE_Result
-stm32_gpio_pm(enum pm_op op, unsigned int pm_hint __unused,
-	      const struct pm_callback_handle *pm_handle __unused)
+stm32_gpio_sec_config_pm(enum pm_op op, unsigned int pm_hint,
+			 const struct pm_callback_handle *hdl __unused)
 {
-	TEE_Result ret = 0;
+	TEE_Result ret = TEE_ERROR_GENERIC;
 
 	if (!IS_ENABLED(CFG_STM32MP13) && !IS_ENABLED(CFG_STM32MP15) &&
 	    pm_hint != PM_HINT_CONTEXT_STATE)
 		return TEE_SUCCESS;
 
 	if (op == PM_OP_RESUME)
-		ret = stm32_gpio_pm_resume();
+		ret = stm32_gpio_sec_config_resume();
 	else
-		ret = stm32_gpio_pm_suspend();
+		ret = stm32_gpio_sec_config_suspend();
 
 	return ret;
 }
-DECLARE_KEEP_PAGER(stm32_gpio_pm);
+DECLARE_KEEP_PAGER(stm32_gpio_sec_config_pm);
 
 /*
  * Several pinctrl nodes can be probed. Their bank will be put in the unique
@@ -1357,9 +1357,12 @@ static TEE_Result stm32_pinctrl_probe(const void *fdt, int node,
 		panic();
 
 	if (!pm_register) {
-		/* Register to PM once for all probed banks */
-		register_pm_core_service_cb(stm32_gpio_pm, NULL,
-					    "stm32-gpio-service");
+		/*
+		 * Register to PM once for all probed banks to restore
+		 * their secure configuration.
+		 */
+		register_pm_core_service_cb(stm32_gpio_sec_config_pm, NULL,
+					    "stm32-gpio-secure-config");
 		pm_register = true;
 	}
 
-- 
2.47.2

